src/lib/zck.c

```diff
@@ -349,6 +349,23 @@ bool ZCK_PUBLIC_API zck_set_ioption(zckCtx *zck, zck_ioption option, ssize_t val
             if(!set_chunk_hash_type(zck, ZCK_HASH_SHA256))
                 return false;
         }
+    } else if(option == ZCK_NO_WRITE) {
+        if(value == 0) {
+            if(zck->no_write == 1) {
+                set_error(zck, "Unable to enable write after it's been disabled");
+                return false;
+            }
+            zck->no_write = 0;
+        } else if(value == 1) {
+            zck->no_write = 1;
+            if(zck->temp_fd) {
+                close(zck->temp_fd);
+                zck->temp_fd = 0;
```

Comment:
(I'll repeat what I said in the issue, 0 is a valid fd number and shouldn't be used to mark temp_fd invalid. Let's use -1 for this... But it probably should be another PR as parts of the code check for 0 and it's out of scope. That aside this looks good at first glance, didn't test. Thanks for the quick fix!)

Reply:
First test is okay, but I ask Matt to test it on its device, because he cold be better reproduce the issue.